.github/workflows/integration-test-flex-suite.yaml

```diff
@@ -14,10 +18,65 @@ jobs:
         go-version: 1.15
 
     - name: setup minikube
-      uses: manusa/actions-setup-minikube@v2.2.0
+      uses: manusa/actions-setup-minikube@v2.3.0
       with:
-        minikube version: 'v1.13.1'
-        kubernetes version: 'v1.19.2'
+        minikube version: 'v1.18.1'
+        kubernetes version: ${{ matrix.kubernetes-versions }}
+        start args: --memory 6g --cpus=2
+
+    - name: check k8s cluster status
+      run: |
+        kubectl cluster-info
+        kubectl get pods -n kube-system
+
+    - name: use local disk
+      run: |
+        sudo swapoff --all --verbose
+        sudo umount /mnt
+        # search for the device since it keeps changing between sda and sdb
+        sudo wipefs --all --force /dev/$(lsblk|awk '/14G/ {print $1}'| head -1)1
+        sudo lsblk
+
+    - name: build rook
+      run: |
+        # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+        GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='ceph' VERSION=0 build
+        docker images
+        docker tag $(docker images|awk '/build-/ {print $1}') rook/ceph:master
+
+    - name: TestCephFlexSuite
+      run: |
+       export DEVICE_FILTER=$(lsblk|awk '/14G/ {print $1}'| head -1)
+       go test -v -timeout 1800s -run CephFlexSuite github.com/rook/rook/tests/integration
+
+    - name: Artifact
+      uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: ceph-flex-suite-artifact
+        path: /home/runner/work/rook/rook/tests/integration/_output/tests/
+
+  TestCephFlexSuite-Master:
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/tags/release-*'
```

Comment:
Why not just the release? Why would we run on master?

Reply:
According to [this](https://github.com/rook/rook/pull/7051#issuecomment-765517036) comment, we want to run for master also.

Reply:
When is triggered a run on master?

Reply:
so github.ref is `The branch or tag ref that triggered the workflow run. For branches this in the format refs/heads/<branch_name>, and for tags it is refs/tags/<tag_name>.` The `if` condition will check if it is `master` or `tags`

Reply:
I think `on: [push, pull_request]` instead of on: [pull_request] is what left here. I'll just go add that.

Reply:
added `on: [push, pull_request]` to `Triggers the workflow on push or pull request events`

Reply:
About `'refs/heads/master'` again, does that mean this will be triggered once the PR merges? I'm still confused. @travisn is the goal to run 5 suites once the PR merges on master?

Reply:
> About `'refs/heads/master'` again, does that mean this will be triggered once the PR merges? I'm still confused. @travisn is the goal to run 5 suites once the PR merges on master?

@leseb  I believe no. Suppose I create a PR and name the branch `abc`(example) then `github.ref` will check `if  github.ref== 'refs/heads/master'`. In this case it is `github.ref == 'refs/heads/abc'` so it will not get inside this if.

And suppose, I pushed to master branch then it will get inside this `github.ref == 'refs/heads/master'`.

(According to my understanding)

Reply:
Ok but no one ever pushes to master directly, unless it's a tag. Hence I'm confused with the `'refs/heads/master'`.